TaskEngine/Tasks/CleanUpElasticIndexTask.cs

```diff
@@ -0,0 +1,88 @@
+﻿using ClassTranscribeDatabase;
+using ClassTranscribeDatabase.Models;
+using CTCommons;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Configuration;
+using System.Threading.Tasks;
+using static ClassTranscribeDatabase.CommonUtils;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+using Nest;
+using Elasticsearch.Net;
+using System;
+using System.Collections.Generic;
+
+namespace TaskEngine.Tasks
+{
+    class CleanUpElasticIndexTask : RabbitMQTask<string>
+    {
+        private readonly ElasticClient _client;
+
+        public CleanUpElasticIndexTask(RabbitMQConnection rabbitMQ,
+            ILogger<CleanUpElasticIndexTask> logger)
+            : base(rabbitMQ, TaskType.CleanUpElasticIndex, logger)
+        {
+            var configuration = CTDbContext.GetConfigurations();
+
+            // initialize elastic client
+            var node = new Uri(configuration.GetValue<string>("ES_CONNECTION_ADDR"));
+            using (var settings = new ConnectionSettings(node))
+            {
+                //settings.DefaultIndex("classTranscribe");
+                _client = new ElasticClient(settings);
+            }
+        }
+        protected async override Task OnConsume(string example, TaskParameters taskParameters, ClientActiveTasks cleanup)
+        {
+            registerTask(cleanup, "CleanUpElasticIndexTask"); // may throw AlreadyInProgress exception
+            _logger.LogInformation("CleanUpElasticIndexTask Starting");
+
+            var result = await _client.Indices.GetAsync(new GetIndexRequest(Indices.All));
+            var indices = result.Indices;
+            foreach (var index in indices)
+            {
+                string index_name = index.Key.ToString();
+
+                // elastic search internal management index starts with '.'
+                if (index_name[0] == '.')
+                {
+                    continue;
+                }
+
+                _logger.LogInformation(index.Key.ToString());
+                string[] parts = index_name.Split("_");
+                var index_string_id = parts[0];
+                var index_string_lang = parts[1];
+                var index_string_time = parts[2];
+
+                // Indices that were created in the last 2 days would not be removed.
+                DateTime createdAt = DateTime.ParseExact(index_string_time, "yyyyMMddHHmmss", null);
+                _logger.LogInformation("Created at: " + createdAt.ToString());
+                DateTime range = DateTime.Now.AddDays(-2);
```

Comment:
Please make this configurable number of minutes in the environment file

Reply:
Done, new env var added.
In .env:
ES_INDEX_TIME_TO_LIVE=2880

In vs_appsettings.json:
"ES_INDEX_TIME_TO_LIVE": "2880"